MCPServer/Sources/ResourceTemplate.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import MCPInterface
+
+/// A template for resources with variable URI patterns
+public struct ResourceTemplate {
```

Comment:
Some of those types seem to be an extension from those in the specs. ie [ResourceTemplate](https://github.com/gsabran/mcp-swift-sdk/blob/main/MCPInterface/Sources/mcp_interfaces/Interface.swift#L503)

Is this used in the PR?